homeassistant/components/rainmachine/__init__.py

```diff
@@ -128,6 +130,39 @@
 }, extra=vol.ALLOW_EXTRA)
 
 
+async def _check_service_permissions(hass, service_call):
+    """Ensure the user of a service call has proper permissions."""
+    if service_call.data.get('entity_id'):
+        rainmachine_entities = service_call.data['entity_id']
+    else:
+        # If the service call doesn't explicitly provide entity IDs, use _all_
```

Comment:
I don't think this is allowed anymore.  My understanding is that service calls need a specific `entity_id` or an `*` to denote all entities.

Reply:
correct.

Reply:
Thanks for the feedback, but I'm confused: @balloob's previous feedback (https://discordapp.com/channels/330944238910963714/330990195199442944/558731762226692096) was that since none of my services currently utilize an entity ID, the correct way to assess proper permissions was to "iterate through all rainmachine entities and see if the user can control any of them".

Reply:
Oh right, that is correct. 

I was confused by the code because it seems like you are dealing with a service call that targets all entities. The code is actually not what I meant. If a service call is targeting the integration (no specific entity), we should allow that if the user can control ANY entity, not all.

Reply:
@bachya @balloob my bad.